src/prime_rl/trainer/config.py

```diff
@@ -41,13 +42,28 @@ class ModelConfig(BaseConfig):
 
 
 class OptimizerConfig(BaseConfig):
-    """Configures the Adam optimizer."""
+    """Configures the Adam optimizer and learning rate scheduler."""
 
     lr: Annotated[float, Field(ge=0)] = 4e-4
     weight_decay: Annotated[float, Field(ge=0)] = 0.01
     betas1: Annotated[float, Field(ge=0)] = 0.9
     betas2: Annotated[float, Field(ge=0)] = 0.99
 
+    # LR Scheduler parameters
+    scheduler: Annotated[SchedulerType, Field(description="Type of learning rate scheduler to use.")] = "constant"
+
+    n_warmup_steps: Annotated[
+        int, Field(ge=0, description="Number of warmup steps for the learning rate scheduler.")
+    ] = 0
+
+    n_final_decay: Annotated[
+        int | None,
+        Field(
+            ge=1,
+            description="Number of steps to decay the learning rate during the final portion of training. If None, will use max_steps from trainer config.",
+        ),
+    ] = None
```

Comment:
any chance you can allign the config name with the one form torchtitan? https://github.com/pytorch/torchtitan/blob/cf30b2902718790cbe91900414c3201b6d7680b0/torchtitan/config/job_config.py#L135